### DISCUSSION TITLE
Allowing renovate to use artifactory behind org certificate

### DISCUSSION THREAD

**Discussion by d-g-n** [renovatebot/renovate#13896] [Request Help] [answered]
Hi all,

I'm trialing renovate, public cloud for now, for a few Scala/SBT projects in our org. I've seen the docs for passing in a username and password to renovate via the config.js file and that's fine. Problem is that our internal artifactory contains a lot of dependencies that renovate really needs to be able to see, but there's also a certificate required for access to said resource. I had a trawl through the docs and prior discussions and couldn't find any way to pass in said cert, does anyone know how this might be possible?

If self-hosting is required to make said cert available, which we may even do anyway to enhance security, that's fine. The below is the output from the log trying to access the private artifactory, url has been changed to org-artifactory intentionally.

Thanks

```
{
  "failedUrl": "https://org-artifactory/artifactory/repo/org/scala-lang/scala-library/maven-metadata.xml",
  "err": {
    "name": "RequestError",
    "code": "EPROTO",
    "timings": {
      "start": 1643385693358,
      "socket": 1643385693359,
      "lookup": 1643385693388,
      "connect": 1643385693579,
      "upload": 1643385694015,
      "error": 1643385694017,
      "phases": {
        "wait": 1,
        "dns": 29,
        "tcp": 191,
        "request": 436,
        "total": 659
      }
    },
    "message": "write EPROTO 140073354086272:error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 40\n",
    "stack": "RequestError: write EPROTO 140073354086272:error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 40\n\n    at ClientRequest.<anonymous> (/home/ubuntu/renovateapp/node_modules/got/dist/source/core/index.js:962:111)\n    at Object.onceWrapper (events.js:520:26)\n    at ClientRequest.emit (events.js:412:35)\n    at ClientRequest.emit (domain.js:475:12)\n    at ClientRequest.origin.emit (/home/ubuntu/renovateapp/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)\n    at TLSSocket.socketErrorListener (_http_client.js:475:9)\n    at TLSSocket.emit (events.js:400:28)\n    at TLSSocket.emit (domain.js:475:12)\n    at emitErrorNT (internal/streams/destroy.js:106:8)\n    at emitErrorCloseNT (internal/streams/destroy.js:74:3)\n    at processTicksAndRejections (internal/process/task_queues.js:82:21)\n    at WriteWrap.onWriteComplete [as oncomplete] (internal/stream_base_commons.js:94:16)",
    "options": {
      "headers": {
        "user-agent": "Renovate Bot (GitHub App 2740)",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://org-artifactory/artifactory/repo/org/scala-lang/scala-library/maven-metadata.xml",
      "hostType": "maven",
      "username": "",
      "password": "",
      "method": "GET",
      "http2": false
    }
  }
}```

**Comment by viceice** [comment, marked as answer]
https://github.com/containerbase/buildpack/blob/main/docs/custom-root-ca.md

**Reply by d-g-n** [reply]
I see, thanks for that, building and maintaining our own renovate with org certs is certainly viable. I'm assuming there's no way to pass the required certs in any way to the github application version of Renovate however?

**Reply by viceice** [reply]
Nope, but you can extend the official renovate docker image and add your cert on top. 

That image can again automatically updated by renovate. 😉

**Reply by d-g-n** [reply]
Hello again, i've been experimenting with this and have tried a few options, but i'm not a node dev so a lot of it is a mystery to me. I've tried to load the certs at runtime using the NODE_EXTRA_CA_CERTS env var, my certs are originally in a p12 format so i've used `openssl pkcs12 -in somecert.p12 -out newfile.pem -nodes` to export it to a password-less pem file which i then pass into that env var, however regardless of what i do i still get the same error, i'd appreciate any and all help

**Reply by viceice** [reply]
You only need the public cert, not the private key

**Reply by d-g-n** [reply]
Assuming the PR eventually gets merged, this has been solved by https://github.com/renovatebot/renovate/pull/13991
